kernel-module/xt_RTPENGINE.c

```diff
@@ -4544,6 +4544,8 @@ static unsigned int rtpengine46(struct sk_buff *skb, struct rtpengine_table *t,
 	if (g->target.dtls && is_dtls(skb))
 		goto skip1;
 	if (g->target.non_forwarding) {
+		if (g->target.do_intercept)
+			goto not_rtp;
 		if (g->target.blackhole)
 			error_nf_action = NF_DROP;
 		goto skip1;
```

Comment:
I think it would be more appropriate to defuse this entire block when `do_intercept` is set, instead of jumping to `not_rtp`. The `not_rtp` label is specifically meant for media that isn't RTP, and having interception active doesn't mean it's not RTP. For the SRTP case for example you'd still want it to be decrypted.

So I think the appropriate code path with both `non_forwarding` and `blackhole` set would be to still go through almost everything below (including decryption and intercept) and then finally just not forward anything and drop the packet

Reply:
Good points. I just pushed a revised commit which hopefully addresses your comments.

I also noticed the stats weren't incrementing on `non_forwarding` and `blackhole` scenarios. By delaying the drop logic to the final step it also solves this problem.

```
cat /proc/rtpengine/0/list
local inet4 192.168.26.246:10000
    stats:               107156 bytes,                  623 packets,                    0 errors
    SSRC in: f915192d
    option: non forwarding
    option: blackhole
local inet4 192.168.26.246:10016
    stats:               107328 bytes,                  624 packets,                    0 errors
    SSRC in: f930ca98
    option: non forwarding
    option: blackhole

```